Automattic/wp-calypso
# Developer Features: Add Tests for Analytics Tracking

Related to #87474

## Proposed Changes

This simplifies/fixes the analytics tracking for Developer Features. I couldn't actually reproduce the incorrect data, but @wongasy reported it here: https://github.com/Automattic/wp-calypso/pull/87474#issuecomment-1958461911.

~~Originally, this wasn't hardcoded because there was no need for an `id` - that was added in the last minute in order to avoid errors when mapping the features list. As the only condition for the `id` is that it's unique, I see no reason why we can't save a bit of code and use it for recording analytics too. That also guarantees no wrong analytics if the link structure changes in future.~~

As per discussion below, this now just adds tests for the existing approach rather than changing it. This ensures that analytics won't break in future.

## Testing Instructions

Run `yarn test-client client/me/developer/features/test/use-handle-click-link.js` and confirm that they pass. You can change the feature slug in the tests locally to confirm that they fail when incorrect.

cc @nightnei, @oswian, @dlind1
